src/components/SelectionPills/__tests__/SelectionPills.spec.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react'
+import renderer from 'react-test-renderer'
+import { StyleRoot } from 'radium'
+import { mount } from 'enzyme'
+import { spy } from 'sinon'
+import SelectionPills from '../SelectionPills'
+
+describe('SelectionPills', () => {
+  const testPills = [
+    {
+      text: 'bananas',
+      id: 'selection-1',
+    },
+    {
+      text: 'arugula',
+      id: 'selection-2',
+    },
+  ]
+
+  it('renders SelectionPills correctly', () => {
+    const tree = renderer
+      .create(
+        <StyleRoot>
+          <SelectionPills
+            pills={testPills}
+            onSelectPill={(e, pill) => {
+              console.log(pill)
+            }}
+            label={'Filter by'}
+          />
+        </StyleRoot>
+      )
+      .toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+
+  it('renders SelectionPills with select all included correctly', () => {
+    const tree = renderer
+      .create(
+        <StyleRoot>
+          <SelectionPills
+            includeSelectAll
+            selectAllLabel="Select all"
+            pills={testPills}
+            onSelectPill={(e, pill) => {
+              console.log(pill)
+            }}
+            label={'Filter by'}
+          />
+        </StyleRoot>
+      )
+      .toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+
+  it('renders SelectionPills with pill attributes set correctly', () => {
+    const attributesPills = [
+      { text: 'bananas', id: 'selection-1', isSelected: true },
+      { text: 'arugula', id: 'selection-2', isDisabled: true },
+      { text: 'apples', id: 'selection-3', style: { button: { marginLeft: '10px' } } },
+    ]
+    const tree = renderer
+      .create(
+        <StyleRoot>
+          <SelectionPills
+            pills={attributesPills}
+            onSelectPill={(e, pill) => {
+              console.log(pill)
+            }}
+            label={'Filter by'}
+          />
+        </StyleRoot>
+      )
+      .toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+
+  it('handles onPillClick correctly', () => {
+    const onSelectPill = spy()
+    const Pills = mount(
+      <StyleRoot>
+        <SelectionPills pills={testPills} onSelectPill={onSelectPill} />
+      </StyleRoot>
+    )
+
+    Pills.find('input')
+      .first()
+      .simulate('change')
+    expect(onSelectPill.called).toBeTruthy()
+    expect(onSelectPill.withArgs(3)).toBeTruthy()
+  })
+
+  it('disables remaining pills when max selected', () => {
+    const selectPills = [
+      { text: 'bananas', id: 'selection-1' },
+      { text: 'arugula', id: 'selection-2', isSelected: true },
+      { text: 'apple', id: 'selection-3' },
+    ]
+    const Pills = mount(
```

Comment:
another style nit - we conventionally call this `wrapper` or `component` so it's not confused with a React element.